source/common/config/grpc_mux_impl.cc

```diff
@@ -216,7 +216,7 @@ void GrpcMuxImpl::onReceiveTrailingMetadata(Http::HeaderMapPtr&& metadata) {
 void GrpcMuxImpl::onRemoteClose(Grpc::Status::GrpcStatus status, const std::string& message) {
   ENVOY_LOG(warn, "gRPC config stream closed: {}, {}", status, message);
   stream_ = nullptr;
-  handleFailure();
+  setRetryTimer();
```

Comment:
@htuch I think calling setRetryTimer() should be sufficient instead of calling handleFailure() which invokes failure callbacks which in turn increments the metrics because nothing is being updated here.